packages/application-extension/src/index.tsx

```diff
@@ -822,13 +822,21 @@ const propertyInspector: JupyterFrontEndPlugin<IPropertyInspectorProvider> = {
   id: '@jupyterlab/application-extension:property-inspector',
   autoStart: true,
   requires: [ILabShell],
+  optional: [ILayoutRestorer],
   provides: IPropertyInspectorProvider,
-  activate: (app: JupyterFrontEnd, labshell: ILabShell) => {
+  activate: (
+    app: JupyterFrontEnd,
+    labshell: ILabShell,
```

Comment:
I think we can remove this dependency and just use `app.shell` since only `shell.add` is being invoked.

Reply:
We need the full `ILabShell` object to access `currentChanged` in the property inspector implementation.

Reply:
Actually, we should just add that signal to `IShell`, there is already a [`currentWidget`](https://github.com/jupyterlab/jupyterlab/blob/d87de688a111010cfba3165b3806498ea5109e20/packages/application/src/frontend.ts#L253).

Reply:
My mistake, it is necessary.